pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -1355,8 +1344,16 @@ func TestPreempt(t *testing.T) {
 	labelKeys := []string{"hostname", "zone", "region"}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			apiObjs := mergeObjs(test.pod, test.pods)
-			client := clientsetfake.NewSimpleClientset(apiObjs...)
+			client := clientsetfake.NewSimpleClientset()
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
+			podInformer := informerFactory.Core().V1().Pods().Informer()
+			test.pod.Namespace = v1.NamespaceDefault
```

Comment:
It's intended to assign a non-nil namespace, as the informer store uses pod name as the key if the namespace is empty:

https://github.com/kubernetes/kubernetes/blob/180af4240c42906f072112e7ac7818f4b3574c5a/staging/src/k8s.io/client-go/tools/cache/store.go#L99-L111

However, podNamespaceLister just concats namespace + "/" + name as the key to fetch Pod:

https://github.com/kubernetes/kubernetes/blob/180af4240c42906f072112e7ac7818f4b3574c5a/staging/src/k8s.io/client-go/listers/core/v1/pod.go#L90-L99

In a real-world case, API has defaulting logic to apply "default" namespace if namespace is nil. But in UT, we should take care of it.

Reply:
It's actually better to apply the namespace directly in the testcases. I will do that either in this PR, or #92571 - depends on which one gets merged first as they conflict with each other.

Reply:
sound like a bug in client-go.